common/crypto/attestation-api/test/verify_evidence_app/test.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "test.h"
+#include <string>
+#include "error.h"
+#include "logging.h"
+#include "test-defines.h"
+#include "test-utils.h"
+
+#include "attestation_tags.h"
+#include "verify-evidence.h"
+
+bool test()
+{
+    uint32_t buffer_length = 1 << 20;
+    char buffer[buffer_length];
+    uint32_t filled_size;
+    std::string jsonevidence;
+    std::string expected_statement;
+    std::string expected_code_id;
+    std::string wrong_expected_statement;
+    std::string wrong_expected_code_id;
+
+    COND2LOGERR(!load_file(EVIDENCE_FILE, buffer, buffer_length, &filled_size),
+        "can't read input evidence " EVIDENCE_FILE);
+    jsonevidence = std::string(buffer, filled_size);
+
+    COND2LOGERR(!load_file(STATEMENT_FILE, buffer, buffer_length, &filled_size),
+        "can't read input statement " STATEMENT_FILE);
+    expected_statement = std::string(buffer, filled_size);
+
+    COND2LOGERR(!load_file(CODE_ID_FILE, buffer, buffer_length, &filled_size),
+        "can't read input code id " CODE_ID_FILE);
+    expected_code_id = std::string(buffer, filled_size);
+
+    wrong_expected_statement = std::string("wrong statement");
```

Comment:
This makes it fail in the json decoding?  A stronger test case might have been to modify the signature?  But i guess we can add such a test case if we ever encounter issues/bugs after the json decoding ...

Reply:
> This makes it fail already in the initial json decoding?

The json does not contain this. The statement is first hashed and then checked against the report data. That's where is fails.

> An additional interesting test case might have been to modify the signature? But i guess we can add such a test case if we ever encounter issues/bugs after the json decoding ...

This is already (if I remember correctly) a test case in the PDO crypto test suite. Here the tests are relative just to the introduced API.